composer/utils/checkpoint.py

```diff
@@ -256,17 +277,50 @@ def _download_checkpoint(
     return composer_states_filepath, extracted_checkpoint_folder, extracted_rank_n
 
 
+def glob_filter(exclude_globs: List[str]) -> Callable[[Dict], None]:
+    """Provides a function which deletes all subparts of a dictionary based on a list of paths."""
+
+    def filter_func(state_dict: Dict) -> None:
+
+        def recur_filter_func(state_dict: Dict, exclude_glob: List[str]):
+            # End of path, delete key in state_dict
+            if len(exclude_glob) == 1:
+                del state_dict[exclude_glob[0]]
+            # Wildcard, descend down all keys
+            elif exclude_glob[0] == "*":
```

Comment:
Instead of manually parsing glob, can you use https://docs.python.org/3/library/fnmatch.html#module-fnmatch to support `?`, `[something]`, and `[!something]` notion?

Reply:
I think if one ends the path in a wildcard, then I presume that should remove all subkeys?

Reply:
In addition,can we support using a wildcard as part of a key (e.g. `conv*` to remove all conv keys)?

Reply:
moving discussion offline